pkg/start/start.go

```diff
@@ -186,13 +167,32 @@ func (o *Options) makeTLSConfig() (*tls.Config, error) {
 	}), nil
 }
 
+// run launches a number of goroutines to handle manifest application,
+// metrics serving, etc.  It continues operating until ctx.Done(),
+// and then attempts a clean shutdown limited by an internal context
+// with a two-minute cap.  It returns after it successfully collects all
+// launched goroutines.
 func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourcelock.ConfigMapLock) {
-	runContext, runCancel := context.WithCancel(ctx)
+	runContext, runCancel := context.WithCancel(ctx) // so we can cancel internally on errors or TERM
 	defer runCancel()
-	shutdownContext, shutdownCancel := context.WithCancel(ctx)
+	shutdownContext, shutdownCancel := context.WithCancel(context.Background()) // extends beyond ctx
 	defer shutdownCancel()
-	errorChannel := make(chan error, 1)
-	errorChannelCount := 0
+	postMainContext, postMainCancel := context.WithCancel(context.Background()) // extends beyond ctx
+	defer postMainCancel()
+
+	ch := make(chan os.Signal, 1)
+	defer func() { signal.Stop(ch) }()
+	signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
+	go func() {
```

Comment:
You need to defer utilruntime.HandleCrash() every goroutine

Reply:
> ...every goroutine you've added in this code change

I don't think I added any.  For example, this code is in master [here][1].  But sure, I can fill in the previously-missing `HandleCrash`...

[1]: https://github.com/openshift/cluster-version-operator/blob/16d3d8465b8e3f3c557becab28763d6cc18829a6/pkg/start/start.go#L148-L161

Reply:
>  But sure, I can fill in the previously-missing HandleCrash...

Added with e064293e -> 4095f1b4, which also rebases onto master.